modules/claims_api/app/workers/claims_api/poa_updater.rb

```diff
@@ -7,19 +7,26 @@ module ClaimsApi
   class PoaUpdater
     include Sidekiq::Worker
 
-    def perform(power_of_attorney_id)
+    def perform(power_of_attorney_id) # rubocop:disable Metrics/MethodLength
       poa_form = ClaimsApi::PowerOfAttorney.find(power_of_attorney_id)
       service = BGS::Services.new(
         external_uid: poa_form.external_uid,
         external_key: poa_form.external_key
       )
+
+      ssn = poa_form.auth_headers['va_eauth_pnid']
+      file_number = service.people.find_by_ssn(ssn)[:file_nbr] # rubocop:disable Rails/DynamicFindBy
```

Comment:
Just jotting this down as a reminder to us.
But we might find that we'll have to come back and add some kind of logic to handle the case where the `FileNumber` returned is `nil`.
It shouldn't really be a problem going forward since you've added the `check_file_number_exists!` to the POST.  So now the POST and the PUT have that protection and no new submissions should even be able to make it here if a `FileNumber` can't be found.  But if we retry some of the past submissions and they still fail, then it's likely because a `FileNumber` wasn't able to be found here.